Android Release OneDrive API not working

Hey guys,

I am using this library for android, and there are no problems with Debug version, but I am receiving errors for release version with enabled minification and proguard

### Expected behavior

I should be able to load folders of user that are presented in oneDrive
```
graphClient.drive().root().children().buildRequest().get(new ICallback<IDriveItemCollectionPage>() {
                    @Override
                    public void success(IDriveItemCollectionPage iDriveItemCollectionPage) {
                        for (DriveItem driveItem : iDriveItemCollectionPage.getCurrentPage()) {
                        }
                    }
                    @Override
                    public void failure(ClientException ex) {
                        logException(ex.getCause());
                    }
};
```


### Actual behavior

I receive 
```
2019-01-18 17:03:02.929 18445-19364/? E/global: DefaultHttpProvider[sendRequestInternal] - 358Error during http request
2019-01-18 17:03:02.929 18445-19364/? E/global: Throwable detail: com.microsoft.graph.core.ClientException: Error during http request
```



### Steps to reproduce the behavior

Build an android app with enabled minification and proguard.
```
-keepclasseswithmembernames class com.microsoft.** { *; }
```

The same issue here. Can not compile with proguard in release.
I have tried a lot of things for three days.

````java
activity.mGraphClient.me().drive().buildRequest().get(callback )
````

> DefaultHttpProvider[sendRequestInternal] - 370Error during http request
Throwable detail: com.microsoft.graph.core.ClientException: Error during http reque

and 

> com.microsoft.graph.core.ClientException: Error during http requ
at com.microsoft.graph.http.DefaultHttpProvider.sendRequestInternal(:368)
at com.microsoft.graph.http.DefaultHttpProvider.access$000(:53)
at com.microsoft.graph.http.DefaultHttpProvider$1.run(:155)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: com.sun.ws.rs.ext.RuntimeDelegateImpl
at javax.ws.rs.ext.RuntimeDelegate.findDelegate(:122)
at javax.ws.rs.ext.RuntimeDelegate.getInstance(:91)
at javax.ws.rs.core.UriBuilder.newInstance(:69)
at javax.ws.rs.core.UriBuilder.fromUri(:80)
at com.microsoft.graph.http.BaseRequest.getRequestUrl(:147)
at com.microsoft.graph.http.DefaultHttpProvider.sendRequestInternal(:239)
com.techniwork.fastsharing W/System.err

Proguard file :

> -keep class com.microsoft.** { *; }
-keep class com.microsoft.**
-keep interface com.microsoft.** { *; }
-keepclasseswithmembernames class com.microsoft.** { *; }

>-keep class com.sun.ws.** { *; }
-keep class com.sun.ws.**
-keep interface com.sun.ws.** { *; }

>-keep class com.sun.jersey.** { *; }

-keep class com.sun.jersey.**
-keep interface com.sun.jersey.** { *; }

> +--- com.microsoft.graph:microsoft-graph:1.4.+ -> 1.4.0
|    +--- org.apache.commons:commons-math3:3.6.1
|    +--- com.google.code.gson:gson:2.8.2
|    +--- com.sun.jersey:jersey-server:1.19.4
|    |    \--- com.sun.jersey:jersey-core:1.19.4
|    |         \--- javax.ws.rs:jsr311-api:1.1.1
|    \--- com.google.guava:guava:20.0 -> 25.1-android
|         +--- com.google.code.findbugs:jsr305:3.0.2
|         +--- org.checkerframework:checker-compat-qual:2.0.0
|         +--- com.google.errorprone:error_prone_annotations:2.1.3
|         +--- com.google.j2objc:j2objc-annotations:1.1
|         \--- org.codehaus.mojo:animal-sniffer-annotations:1.14

Thank you for taking this into account and help me find a solution ... I am desperate ..

The same issue when enable proguard in release build

proguard rules:

    # Microsoft Graph API
    -keepnames class com.microsoft.** implements java.io.Serializable
    -keepnames class com.microsoft.graph.models.** { *; }
    -keepnames class com.microsoft.identity.** { *; }

    -keep class com.microsoft.** implements java.io.Serializable {
        !private <init>();
    }

    -keep class com.microsoft.graph.models.** {
        !private <init>();
    }

    -keep class com.microsoft.identity.** {
        !private <init>();
    }

Crash:

    com.microsoft.graph.core.ClientException: Error during http request
        at f.z.b.d.h.a(CoreHttpProvider.java:421)
        at f.z.b.d.h.a(CoreHttpProvider.java:204)
        at f.z.b.d.h.a(CoreHttpProvider.java:184)
        at f.z.b.d.c.i(BaseCollectionRequest.java:89)
        at f.z.b.h.a.j2.j(DriveItemCollectionRequest.java:52)
        at f.z.b.h.a.i2.run(DriveItemCollectionRequest.java:43)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: com.sun.ws.rs.ext.RuntimeDelegateImpl
        at l.b.a.b.c.a(RuntimeDelegate.java:122)
        at l.b.a.b.c.b(RuntimeDelegate.java:91)
        at l.b.a.a.a.a(UriBuilder.java:69)
        at l.b.a.a.a.a(UriBuilder.java:80)
        at f.z.b.d.d.b(BaseRequest.java:176)
        at f.z.b.d.c.b(BaseCollectionRequest.java:112)
        at f.z.b.d.h.a(CoreHttpProvider.java:237)
        at f.z.b.d.h.a(CoreHttpProvider.java:204) 
        at f.z.b.d.h.a(CoreHttpProvider.java:184) 
        at f.z.b.d.c.i(BaseCollectionRequest.java:89) 
        at f.z.b.h.a.j2.j(DriveItemCollectionRequest.java:52) 
        at f.z.b.h.a.i2.run(DriveItemCollectionRequest.java:43) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636) 
        at java.lang.Thread.run(Thread.java:764) 
     Caused by: java.lang.ClassNotFoundException: com.sun.ws.rs.ext.RuntimeDelegateImpl
        at java.lang.Class.classForName(Native Method)
        at java.lang.Class.forName(Class.java:453)
        at java.lang.Class.forName(Class.java:378)
        at l.b.a.b.b.a(FactoryFinder.java:62)
        at l.b.a.b.b.a(FactoryFinder.java:155)
        at l.b.a.b.c.a(RuntimeDelegate.java:105)
        at l.b.a.b.c.b(RuntimeDelegate.java:91) 
        at l.b.a.a.a.a(UriBuilder.java:69) 
        at l.b.a.a.a.a(UriBuilder.java:80) 
        at f.z.b.d.d.b(BaseRequest.java:176) 
        at f.z.b.d.c.b(BaseCollectionRequest.java:112) 
        at f.z.b.d.h.a(CoreHttpProvider.java:237) 
        at f.z.b.d.h.a(CoreHttpProvider.java:204) 
        at f.z.b.d.h.a(CoreHttpProvider.java:184) 
        at f.z.b.d.c.i(BaseCollectionRequest.java:89) 
        at f.z.b.h.a.j2.j(DriveItemCollectionRequest.java:52) 
        at f.z.b.h.a.i2.run(DriveItemCollectionRequest.java:43) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636) 
        at java.lang.Thread.run(Thread.java:764) 
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.sun.ws.rs.ext.RuntimeDelegateImpl" on path: DexPathList[[zip file "/data/app/com.flowiemusic.tiles.mp3.player.magictiles-5vKwBPcf-f3yF414EgNZqw==/base.apk"],nativeLibraryDirectories=[/data/app/com.flowiemusic.tiles.mp3.player.magictiles-5vKwBPcf-f3yF414EgNZqw==/lib/x86, /data/app/com.flowiemusic.tiles.mp3.player.magictiles-5vKwBPcf-f3yF414EgNZqw==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:125)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at java.lang.Class.classForName(Native Method) 
        at java.lang.Class.forName(Class.java:453) 
        at java.lang.Class.forName(Class.java:378) 
        at l.b.a.b.b.a(FactoryFinder.java:62) 
        at l.b.a.b.b.a(FactoryFinder.java:155) 
        at l.b.a.b.c.a(RuntimeDelegate.java:105) 
        at l.b.a.b.c.b(RuntimeDelegate.java:91) 
        at l.b.a.a.a.a(UriBuilder.java:69) 
        at l.b.a.a.a.a(UriBuilder.java:80) 
        at f.z.b.d.d.b(BaseRequest.java:176) 
        at f.z.b.d.c.b(BaseCollectionRequest.java:112) 
        at f.z.b.d.h.a(CoreHttpProvider.java:237) 
        at f.z.b.d.h.a(CoreHttpProvider.java:204) 
        at f.z.b.d.h.a(CoreHttpProvider.java:184) 
        at f.z.b.d.c.i(BaseCollectionRequest.java:89) 
        at f.z.b.h.a.j2.j(DriveItemCollectionRequest.java:52) 
        at f.z.b.h.a.i2.run(DriveItemCollectionRequest.java:43) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636) 
        at java.lang.Thread.run(Thread.java:764) 

--- a/build.gradle
+++ b/build.gradle
@@ -33,7 +33,7 @@ dependencies {
     
     implementation 'com.google.code.gson:gson:2.8.6'
     
-    implementation 'com.sun.jersey:jersey-server:1.19.4'
+    implementation 'com.squareup.okhttp3:okhttp:3.12.1'
     
     // Core Http library 
     implementation 'com.microsoft.graph:microsoft-graph-core:1.0.2'

--- a/src/main/java/com/microsoft/graph/http/BaseRequest.java
+++ b/src/main/java/com/microsoft/graph/http/BaseRequest.java
@@ -23,7 +23,9 @@
 package com.microsoft.graph.http;
 
 import java.net.URI;
-import javax.ws.rs.core.UriBuilder;
+
+import okhttp3.HttpUrl;
+import okhttp3.HttpUrl.Builder;
 
 import com.microsoft.graph.concurrency.ICallback;
 import com.microsoft.graph.core.IBaseClient;
@@ -172,11 +174,10 @@ public abstract class BaseRequest implements IHttpRequest {
     @Override
     public URL getRequestUrl() {
         String requestUrl = addFunctionParameters();
-        URI baseUrl = URI.create(requestUrl);
-        final UriBuilder uriBuilder = UriBuilder.fromUri(baseUrl);
+        final Builder uriBuilder = HttpUrl.parse(requestUrl).newBuilder();
 
         for (final QueryOption option : queryOptions) {
-        	uriBuilder.queryParam(option.getName(), option.getValue().toString());
+        	uriBuilder.addQueryParameter(option.getName(), option.getValue().toString());
         }
 
         try {

--- a/src/main/java/com/microsoft/graph/http/BaseRequest.java
+++ b/src/main/java/com/microsoft/graph/http/BaseRequest.java
@@ -22,8 +22,6 @@
 
 package com.microsoft.graph.http;
 
-import java.net.URI;
-
 import okhttp3.HttpUrl;
 import okhttp3.HttpUrl.Builder;
 

--- a/src/test/java/com/microsoft/graph/http/BaseCollectionPageTests.java
+++ b/src/test/java/com/microsoft/graph/http/BaseCollectionPageTests.java
@@ -24,7 +24,7 @@ public class BaseCollectionPageTests {
     private IRequestBuilder mRequestBuilder;
     private static ArrayList<String> list;
     private BaseCollectionPage<String, IRequestBuilder> baseCollectionPage;
-    private String requestUrl = "https://a.b.c";
+    private String requestUrl = "https://a.b.c/";
 
     @Before
     public void setUp() throws Exception {

--- a/src/test/java/com/microsoft/graph/http/BaseCollectionRequestTests.java
+++ b/src/test/java/com/microsoft/graph/http/BaseCollectionRequestTests.java
@@ -59,7 +59,7 @@ public class BaseCollectionRequestTests {
                 new MockLogger());
         mProvider.setConnectionFactory(new MockConnectionFactory(new MockConnection(data)));
         mBaseClient.setHttpProvider(mProvider);
-        request = new BaseCollectionRequest<JsonObject,String>("https://a.b.c", mBaseClient, null, JsonObject.class,null){};
+        request = new BaseCollectionRequest<JsonObject,String>("https://a.b.c/", mBaseClient, null, JsonObject.class,null){};
     }
 
     @Test
@@ -80,11 +80,11 @@ public class BaseCollectionRequestTests {
     public void testFunctionParameters() {
         final Option f1 = new FunctionOption("1", "one");
         final Option f2 = new FunctionOption("2", null);
-        final BaseCollectionRequest<String,String> request = new BaseCollectionRequest<String,String>("https://a.b.c", null, Arrays.asList(f1, f2), null,null){};
+        final BaseCollectionRequest<String,String> request = new BaseCollectionRequest<String,String>("https://a.b.c/", null, Arrays.asList(f1, f2), null,null){};
         String urlTest = request.getRequestUrl().toString();
-        assertEquals("https://a.b.c(1='one',2=null)", request.getRequestUrl().toString());
+        assertEquals("https://a.b.c/(1='one',2=null)", request.getRequestUrl().toString());
         request.addFunctionOption(new FunctionOption("3","two"));;
-        assertEquals("https://a.b.c(1='one',2=null,3='two')", request.getRequestUrl().toString());
+        assertEquals("https://a.b.c/(1='one',2=null,3='two')", request.getRequestUrl().toString());
         assertEquals(4, request.getOptions().size());
     }
 
@@ -92,10 +92,10 @@ public class BaseCollectionRequestTests {
     public void testQueryParameters() {
         final Option q1 = new QueryOption("q1","option1 ");
         final Option q2 = new QueryOption("q2","option2");
-        final BaseCollectionRequest<String,String> request = new BaseCollectionRequest<String,String>("https://a.b.c", null, Arrays.asList(q1, q2), null,null){};
-        assertEquals("https://a.b.c?q1=option1+&q2=option2", request.getRequestUrl().toString());
+        final BaseCollectionRequest<String,String> request = new BaseCollectionRequest<String,String>("https://a.b.c/", null, Arrays.asList(q1, q2), null,null){};
+        assertEquals("https://a.b.c/?q1=option1%20&q2=option2", request.getRequestUrl().toString());
         request.addQueryOption(new QueryOption("q3","option3"));
-        assertEquals("https://a.b.c?q1=option1+&q2=option2&q3=option3", request.getRequestUrl().toString());
+        assertEquals("https://a.b.c/?q1=option1%20&q2=option2&q3=option3", request.getRequestUrl().toString());
         assertEquals(4,request.getOptions().size());
     }
 
@@ -105,8 +105,8 @@ public class BaseCollectionRequestTests {
         final Option f2 = new FunctionOption("f2", null);
         final Option q1 = new QueryOption("q1","option1 ");
         final Option q2 = new QueryOption("q2","option2");
-        final BaseCollectionRequest<String,String> request = new BaseCollectionRequest<String,String>("https://a.b.c", null, Arrays.asList(f1, f2, q1, q2), null,null){};
-        assertEquals("https://a.b.c(f1='fun1',f2=null)?q1=option1+&q2=option2", request.getRequestUrl().toString());
+        final BaseCollectionRequest<String,String> request = new BaseCollectionRequest<String,String>("https://a.b.c/", null, Arrays.asList(f1, f2, q1, q2), null,null){};
+        assertEquals("https://a.b.c/(f1='fun1',f2=null)?q1=option1%20&q2=option2", request.getRequestUrl().toString());
         assertEquals(5, request.getOptions().size());
     }
 
@@ -123,7 +123,7 @@ public class BaseCollectionRequestTests {
     public void testHeader() {
         String expectedHeader = "header key";
         String expectedValue = "header value";
-        final BaseCollectionRequest<String,String> request = new BaseCollectionRequest<String,String>("https://a.b.c", null, null, null,null){};
+        final BaseCollectionRequest<String,String> request = new BaseCollectionRequest<String,String>("https://a.b.c/", null, null, null,null){};
         assertEquals(1, request.getHeaders().size());
         request.addHeader(expectedHeader,expectedValue);
         assertEquals(2,request.getHeaders().size());

--- a/src/test/java/com/microsoft/graph/http/BaseRequestBuilderTests.java
+++ b/src/test/java/com/microsoft/graph/http/BaseRequestBuilderTests.java
@@ -11,7 +11,7 @@ import org.junit.Test;
  */
 public class BaseRequestBuilderTests {
 
-    private String expectedRequestUrl = "https://a.b.c";
+    private String expectedRequestUrl = "https://a.b.c/";
     private BaseRequestBuilder baseRequestBuilder;
 
     @Before

--- a/src/test/java/com/microsoft/graph/http/BaseRequestTests.java
+++ b/src/test/java/com/microsoft/graph/http/BaseRequestTests.java
@@ -64,7 +64,7 @@ public class BaseRequestTests {
                 new MockLogger());
         mProvider.setConnectionFactory(new MockConnectionFactory(new MockConnection(data)));
         mBaseClient.setHttpProvider(mProvider);
-        request = new BaseRequest("https://a.b.c", mBaseClient, null,null){};
+        request = new BaseRequest("https://a.b.c/", mBaseClient, null,null){};
     }
 
     @Test
@@ -102,10 +102,10 @@ public class BaseRequestTests {
     public void testFunctionParameters() {
         final Option fo1 = new FunctionOption("1", "one");
         final Option fo2 = new FunctionOption("2", null);
-        final BaseRequest request = new BaseRequest("https://a.b.c", null, Arrays.asList(fo1, fo2), null){};
-        assertEquals("https://a.b.c(1='one',2=null)", request.getRequestUrl().toString());
+        final BaseRequest request = new BaseRequest("https://a.b.c/", null, Arrays.asList(fo1, fo2), null){};
+        assertEquals("https://a.b.c/(1='one',2=null)", request.getRequestUrl().toString());
         request.addFunctionOption(new FunctionOption("3","two"));;
-        assertEquals("https://a.b.c(1='one',2=null,3='two')", request.getRequestUrl().toString());
+        assertEquals("https://a.b.c/(1='one',2=null,3='two')", request.getRequestUrl().toString());
         assertEquals(4, request.getOptions().size());
     }
 
@@ -113,10 +113,10 @@ public class BaseRequestTests {
     public void testQueryParameters() {
         final Option q1 = new QueryOption("q1","option1 ");
         final Option q2 = new QueryOption("q2","option2");
-        final BaseRequest request = new BaseRequest("https://a.b.c", null, Arrays.asList(q1, q2), null){};
-        assertEquals("https://a.b.c?q1=option1+&q2=option2", request.getRequestUrl().toString());
+        final BaseRequest request = new BaseRequest("https://a.b.c/", null, Arrays.asList(q1, q2), null){};
+        assertEquals("https://a.b.c/?q1=option1%20&q2=option2", request.getRequestUrl().toString());
         request.addQueryOption(new QueryOption("q3","option3"));
-        assertEquals("https://a.b.c?q1=option1+&q2=option2&q3=option3", request.getRequestUrl().toString());
+        assertEquals("https://a.b.c/?q1=option1%20&q2=option2&q3=option3", request.getRequestUrl().toString());
         assertEquals(4,request.getOptions().size());
     }
 
@@ -126,14 +126,14 @@ public class BaseRequestTests {
         final Option f2 = new FunctionOption("f2", null);
         final Option q1 = new QueryOption("q1","option1 ");
         final Option q2 = new QueryOption("q2","option2");
-        final BaseRequest request = new BaseRequest("https://a.b.c", null, Arrays.asList(f1, f2, q1, q2), null){};
-        assertEquals("https://a.b.c(f1='fun1',f2=null)?q1=option1+&q2=option2", request.getRequestUrl().toString());
+        final BaseRequest request = new BaseRequest("https://a.b.c/", null, Arrays.asList(f1, f2, q1, q2), null){};
+        assertEquals("https://a.b.c/(f1='fun1',f2=null)?q1=option1%20&q2=option2", request.getRequestUrl().toString());
         assertEquals(5, request.getOptions().size());
     }
 
     @Test
     public void testHttpMethod() {
-        final BaseRequest request = new BaseRequest("https://a.b.c", null, null, null){};
+        final BaseRequest request = new BaseRequest("https://a.b.c/", null, null, null){};
         assertNull(request.getHttpMethod());
         request.setHttpMethod(HttpMethod.GET);
         assertEquals(HttpMethod.GET, request.getHttpMethod());
@@ -143,7 +143,7 @@ public class BaseRequestTests {
     public void testHeader() {
         String expectedHeader = "header key";
         String expectedValue = "header value";
-        final BaseRequest request = new BaseRequest("https://a.b.c", null, null, null){};
+        final BaseRequest request = new BaseRequest("https://a.b.c/", null, null, null){};
         assertEquals(1, request.getHeaders().size());
         assertEquals("SdkVersion", request.getHeaders().get(0).getName());
         //assertEquals(String.format("graph-android-v%s", BuildConfig.VERSION_NAME), request.getHeaders().get(0).getValue());
@@ -157,7 +157,7 @@ public class BaseRequestTests {
         assertEquals(0, request.queryOptions.size());
         final Option q1 = new QueryOption("q1","option1 ");
         final Option f1 = new FunctionOption("f1","option2");
-        final BaseRequest request = new BaseRequest("https://a.b.c", null, Arrays.asList(q1,f1), null){};
+        final BaseRequest request = new BaseRequest("https://a.b.c/", null, Arrays.asList(q1,f1), null){};
         assertEquals(1, request.functionOptions.size());
         assertEquals(1, request.queryOptions.size());
         assertEquals("q1", request.queryOptions.get(0).getName());

--- a/src/test/java/com/microsoft/graph/http/BaseStreamRequestTests.java
+++ b/src/test/java/com/microsoft/graph/http/BaseStreamRequestTests.java
@@ -61,7 +61,7 @@ public class BaseStreamRequestTests {
                 new MockLogger());
         mProvider.setConnectionFactory(new MockConnectionFactory(new MockConnection(data)));
         mBaseClient.setHttpProvider(mProvider);
-        final BaseStreamRequest<String> request = new BaseStreamRequest<String>("https://a.b.c", mBaseClient,null, null){};
+        final BaseStreamRequest<String> request = new BaseStreamRequest<String>("https://a.b.c/", mBaseClient,null, null){};
         request.send();
         assertEquals(1, mAuthenticationProvider.getInterceptionCount());
     }
@@ -105,7 +105,7 @@ public class BaseStreamRequestTests {
                 failure.set(true);
             }
         };
-        final BaseStreamRequest<InputStream> request = new BaseStreamRequest<InputStream>("https://a.b.c", mBaseClient,null, InputStream.class){};
+        final BaseStreamRequest<InputStream> request = new BaseStreamRequest<InputStream>("https://a.b.c/", mBaseClient,null, InputStream.class){};
         request.send(callback);
         assertTrue(success.get());
         assertFalse(failure.get());
@@ -151,7 +151,7 @@ public class BaseStreamRequestTests {
                 failure.set(true);
             }
         };
-        final BaseStreamRequest<InputStream> request = new BaseStreamRequest<InputStream>("https://a.b.c", mBaseClient,null, InputStream.class){};
+        final BaseStreamRequest<InputStream> request = new BaseStreamRequest<InputStream>("https://a.b.c/", mBaseClient,null, InputStream.class){};
         request.send(new byte[]{1, 2, 3, 4},callback);
         assertTrue(success.get());
         assertFalse(failure.get());
@@ -184,15 +184,15 @@ public class BaseStreamRequestTests {
                 new MockLogger());
         mProvider.setConnectionFactory(new MockConnectionFactory(new MockConnection(data)));
         mBaseClient.setHttpProvider(mProvider);
-        final BaseStreamRequest<InputStream> request = new BaseStreamRequest<InputStream>("https://a.b.c", mBaseClient,null, InputStream.class){};
+        final BaseStreamRequest<InputStream> request = new BaseStreamRequest<InputStream>("https://a.b.c/", mBaseClient,null, InputStream.class){};
         request.send(new byte[]{1, 2, 3, 4});
         assertEquals(1, mAuthenticationProvider.getInterceptionCount());
     }
 
     @Test
     public void testBaseMethod() {
-        final BaseStreamRequest<InputStream> request = new BaseStreamRequest<InputStream>("https://a.b.c", mBaseClient,null, InputStream.class){};
-        assertEquals("https://a.b.c", request.getRequestUrl().toString());
+        final BaseStreamRequest<InputStream> request = new BaseStreamRequest<InputStream>("https://a.b.c/", mBaseClient,null, InputStream.class){};
+        assertEquals("https://a.b.c/", request.getRequestUrl().toString());
         request.addHeader("header key", "header value");
         assertEquals(2,request.getHeaders().size());
         assertNull(request.getHttpMethod());

